src/transactions/test/LiquidityPoolTradeTests.cpp

```diff
@@ -0,0 +1,322 @@
+                auto share =
+                    toPoolAsset < fromPoolAsset
+                        ? makeChangeTrustAssetPoolShare(toPoolAsset,
+                                                        fromPoolAsset,
+                                                        LIQUIDITY_POOL_FEE_V18)
+                        : makeChangeTrustAssetPoolShare(fromPoolAsset,
+                                                        toPoolAsset,
+                                                        LIQUIDITY_POOL_FEE_V18);
+                auto pool = xdrSha256(share.liquidityPool());
+
+                int64_t toPoolBalanceBefore1 = getBalance(a1, toPoolAsset);
+                int64_t fromPoolBalanceBefore1 = getBalance(a1, fromPoolAsset);
+                int64_t toPoolBalanceBefore2 = getBalance(a2, toPoolAsset);
+                int64_t fromPoolBalanceBefore2 = getBalance(a2, fromPoolAsset);
+                if (toPoolAsset.type() == ASSET_TYPE_NATIVE)
+                {
+                    toPoolBalanceBefore2 -= 100;
+                }
+                else if (fromPoolAsset.type() == ASSET_TYPE_NATIVE)
+                {
+                    fromPoolBalanceBefore2 -= 100;
+                }
+
+                auto const res = apply();
+                REQUIRE(res.success().offers.size() == 1);
+                REQUIRE(res.success().offers[0].liquidityPool() ==
+                        ClaimLiquidityAtom(pool, fromPoolAsset, fromPool,
+                                           toPoolAsset, toPool));
+
+                REQUIRE(getBalance(a1, toPoolAsset) == toPoolBalanceBefore1);
+                REQUIRE(getBalance(a1, fromPoolAsset) ==
+                        fromPoolBalanceBefore1);
+                REQUIRE(getBalance(a2, toPoolAsset) ==
+                        toPoolBalanceBefore2 - toPool);
+                REQUIRE(getBalance(a2, fromPoolAsset) ==
+                        fromPoolBalanceBefore2 + fromPool);
+
+                LedgerTxn ltx(app->getLedgerTxnRoot());
+                auto lp = loadLiquidityPool(ltx, pool);
+                auto const& cp =
+                    lp.current().data.liquidityPool().body.constantProduct();
+                if (fromPoolAsset == cp.params.assetA &&
+                    toPoolAsset == cp.params.assetB)
+                {
+                    REQUIRE(cp.reserveA == 1000 - fromPool);
+                    REQUIRE(cp.reserveB == 1000 + toPool);
+                }
+                else if (fromPoolAsset == cp.params.assetB &&
+                         toPoolAsset == cp.params.assetA)
+                {
+                    REQUIRE(cp.reserveA == 1000 + toPool);
+                    REQUIRE(cp.reserveB == 1000 - fromPool);
+                }
+                else
+                {
+                    REQUIRE(false);
+                }
+            };
+
+            SECTION("both non-native, strict receive")
+            {
+                SECTION("pool sells cur1 satisfying limit")
```

Comment:
you should probably refactor those tests so that you can easily swap cur1/cur2 instead of copy/pasting sections